### ISSUE TITLE
Using beast 1.83.0.bzl.1 fails to include boost/beast/core/impl/error.ipp

### ISSUE THREAD

**Issue by amarin15** [bazelboost/beast#1]
Trying to build a bazel 7.0 project on Mac OS Sonoma 14.2 using the following:
- `MODULE.bazel` contains
```
bazel_dep(name = "boost.asio", version = "1.83.0.bzl.3")
bazel_dep(name = "boost.beast", version = "1.83.0.bzl.1")
```
- `BUILD.bazel` contains a `cc_library` with
```
deps = [
    "@boost.asio",
    "@boost.beast",
],
```
- building with just `#include <boost/asio.hpp>` works fine
- building with `#include <boost/beast.hpp>` fails with
```
external/boost.beast~1.83.0.bzl.1/include/boost/beast/core/error.hpp:85:10: fatal error: 'boost/beast/core/impl/error.ipp' file not found
```
I can see the `error.ipp` file does exist and has read permissions under `[bazel_folder]/6194c68004852dd0d680e05a4da3a80e/external/boost.beast~1.83.0.bzl.1/include/boost/beast/core/impl` 
- The `error.ipp` include is coming from [here](https://github.com/boostorg/beast/blob/b1779ced98b40bcbd5101bc70b6e0eb233c545df/include/boost/beast/http/error.hpp#L175-L176)
- using `#include <boost/beast/core/impl/error.ipp>` directly also fails to find the `error.ipp` file
- using `#include <boost/beast/core/impl/error.hpp>` finds the `error.hpp` file, but fails with other build errors
- building [`boost::beast`](https://github.com/boostorg/beast) from source without bazel works fine
- `boost::asio` also seems to include some `.ipp` files, so it doesn't look like it's an issue with bazel not finding `.ipp` files in general

Any thoughts on where this error might be coming from?

**Comment by zaucy** [comment]
thanks for the report! I've moved your issue to `bazelboost/beast`. Some bazel boost modules are more incomplete than others. When I get a chance I'll add tests and CI to the repository.

**Comment by amarin15** [comment]
@zaucy thanks!

Taking a look at the differences between the `bazelboost` changes [for beast](https://github.com/boostorg/beast/compare/develop...bazelboost:beast:bazelboost-1.83.0) and [for asio](https://github.com/boostorg/asio/compare/develop...bazelboost:asio:bazelboost-1.83.0), I believe [this change](https://github.com/bazelboost/beast/pull/2) might help, although I am not sure it is sufficient, so it would need testing.


Based on [the logic throwing the error](https://github.com/boostorg/beast/blob/b1779ced98b40bcbd5101bc70b6e0eb233c545df/include/boost/beast/http/error.hpp#L175-L176) which uses `BOOST_BEAST_HEADER_ONLY` and [this logic in config.hpp](https://github.com/boostorg/beast/blob/develop/include/boost/beast/core/detail/config.hpp#L71-L77) which defines `BOOST_BEAST_HEADER_ONLY`, it might be tempting to try the following workaround:

`#define BOOST_BEAST_SEPARATE_COMPILATION 1`
before
`#include <boost/beast.hpp>`

However, there are certain definitions (such as [beast::http::detail::trim](https://github.com/bazelboost/beast/blob/3f741e392f96a17e1971ca082b89c8fb93151ba0/include/boost/beast/http/detail/rfc7230.ipp#L242-L259)) which are found exclusively in `ipp` files, so building will fail with an `Undefined symbols` error in those cases.

**Comment by amarin15** [comment]
Tested that my build succeeds with a [git_override](https://bazel.build/rules/lib/globals/module#git_override) to [this fix](https://github.com/bazelboost/beast/pull/2) and fails without.

**Comment by zaucy** [comment]
Thanks for the patch and testing it out! I published `bazelboost-1.83.0.bzl.2` with your changes. Should be available on the registry shortly.

I'm currently away so it's a bit difficult to test but hopefully it works for you :)

**Comment by amarin15** [comment]
@zaucy thanks! I can confirm that this issue no longer happens when using `1.83.0.bzl.2`.
